### DISCUSSION TITLE
Fundamental questions on kubescape operator on kubernetes

### DISCUSSION THREAD

**Discussion by jaiprasad266** [kubescape/kubescape#839] [Q&A] [answered]
1) Does kubescape-scheduler cron job trigger "configuration scan" by internally firing /v1/scan endpoint and by default scan all the namespaces ?

> If yes , after kubescape-scheduler completes scanning and generates the report and submit to the armo cloud account , how can I download it ?
> 
> What am I looking for/want to do ?
> 
> Schedule the kubescape-scheduler cron job to run every midnight and then once it completes the scanning , I want the report in my github repository.

2) I understand that kubevuln-scheduler crob job does a scanning for image vulnerabilites in the cluster and uploads the report to armo cloud.

> How can I download this report ? I do not want to download manually from the portal , how can I automate this as part of github actions and download it to github repo.

3) On similar lines how to download the rbac visualizer as a file ?

Why we want to do this ?

Basically we want to schedule the cron to scan  and then have a evidence report for each scan in a github repo. 

Reference to Document followed:

https://hub.armosec.io/docs/installation-of-armo-in-cluster

**Comment by Bezbran** [comment, marked as answer]
@jaiprasad266 ,
I will just write here what I wrote in [discord](https://discord.com/channels/893048809884643379/1011507299333980261/1022045270055325727), for the sake of the next people coming with the same need:

You can generate an API token in Kubescape cloud. Then you can use it to access the Kubescape cloud API directly.
Follow the "how to get a cookie" instructions in [Kubescape Cloud swagger](https://api.armosec.io/openapi/v2/swaggerui).

You can store the API token as a Github secret for example.

**Reply by jaiprasad266** [reply]
Thank you @Bezbran , this helps.

**Reply by jaiprasad266** [reply]
@Bezbran 
getting this error while trying to fetch bearer token
```{"errors":["We're facing some difficulties. Please try again"]}```

**Reply by jaiprasad266** [reply]
I was not able to use the kubescape cloud api as I am blocked with fetching the bear token for authentication.

**Comment by amirkCA** [comment]
@jaiprasad266 My name is Amir, I am the VP Product@ARMO. You can send the scanned reports to a Slack channel. Look at settings-> integrations. LMK if you want something similar to Github. 
At this point, it is for configuration scanning, not image scanning. 
Feel free to contact me directly amirk@armosec.io

**Reply by jaiprasad266** [reply]
@amirkCA yes want to store the scanned report in github.
